app/models/donation.rb

```diff
@@ -44,4 +41,32 @@ def value
   def item_count
     donation_details.map(&:quantity).sum
   end
+
+  private
+
+  def skip_adding_donations?(params, required)
+    return false if required
+    return true if params[:donation].blank?
+    return true if params[:donation][:donation_details].blank?
+    return true if params[:donation][:donation_details][:item_id].blank?
+    false
+  end
+
+  def add_to_donation!(params, required = false)
```

Comment:
I find it slightly easier to understand calls to this function using keyword arguments. 
```
add_to_donation!(params, required: false)
```
And in cases when you want to explicity make it required `add_to_donation!(params, required: true)` otherwise it will be false with the default.

Reply:
Excellent suggestion.